File: 백준/Silver/20436. ZOAC 3/ZOAC 3.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        // 키보드 배열 정의
        String[] keyboard = {
            "qwertyuiop",
            "asdfghjkl",
            "zxcvbnm"
        };
    
        // 각 키의 위치를 저장하기 위한 맵
        Map<Character, int[]> positions = new HashMap<>();
        for (int r = 0; r < keyboard.length; r++) {
            for (int c = 0; c < keyboard[r].length(); c++) {
                positions.put(keyboard[r].charAt(c), new int[]{r, c});
            }
        }

        // 초기 손가락 위치를 입력받음
        String[] startPositions = br.readLine().trim().split(" ");
        char leftStart = startPositions[0].charAt(0);
        char rightStart = startPositions[1].charAt(0);

        // 입력할 단어를 받음
        String word = br.readLine().trim();

        // 손가락 초기 위치 설정
        int[] leftPos = positions.get(leftStart);
        int[] rightPos = positions.get(rightStart);

        // 총 시간 초기화 (각 키를 누르는 시간은 1초)
        int totalTime = 0;

        // 주어진 단어를 입력하는 과정에서 시간을 계산
        for (char charInWord : word.toCharArray()) {
            int[] targetPos = positions.get(charInWord);
            // 모음과 자음에 따라 어느 손가락으로 이동할지 결정
            if ("qwertasdfgzxcv".indexOf(charInWord) != -1) { // 자음은 왼손으로
                totalTime += Math.abs(leftPos[0] - targetPos[0]) + Math.abs(leftPos[1] - targetPos[1]) + 1;
                leftPos = targetPos; // 왼손 위치 업데이트
            } else { // 모음은 오른손으로
                totalTime += Math.abs(rightPos[0] - targetPos[0]) + Math.abs(rightPos[1] - targetPos[1]) + 1;
                rightPos = targetPos; // 오른손 위치 업데이트
            }
        }

        // 총 시간을 출력
        System.out.println(totalTime);
    }
}

File: 백준/Silver/20436. ZOAC 3/ZOAC 3.py
import sys

input = sys.stdin.readline

# 키보드 배열을 정의합니다.
keyboard = [
    "qwertyuiop",
    "asdfghjkl",
    "zxcvbnm"
]

# 각 키의 위치를 저장합니다.
positions = {}
for r, row in enumerate(keyboard):
    for c, char in enumerate(row):
        positions[char] = (r, c)

# 초기 손가락 위치를 입력받습니다.
left_start, right_start = input().strip().split()
word = input().strip()

# 손가락 위치를 초기화합니다.
left_pos = positions[left_start]
right_pos = positions[right_start]

# 총 시간 초기화 (각 키를 누르는 시간은 1초)
total_time = 0

# 주어진 단어를 입력하는 과정에서 시간을 계산합니다.
for char in word:
    r, c = positions[char]
    # 모음과 자음에 따라 어느 손가락으로 이동할지 결정합니다.
    if char in "qwertasdfgzxcv":  # 자음은 왼손으로
        total_time += abs(left_pos[0] - r) + abs(left_pos[1] - c) + 1
        left_pos = (r, c)
    else:  # 모음은 오른손으로
        total_time += abs(right_pos[0] - r) + abs(right_pos[1] - c) + 1
        right_pos = (r, c)

print(total_time)